ludwig/schema/hyperopt/scheduler.py

```diff
@@ -107,6 +101,19 @@ class BaseSchedulerConfig(schema_utils.BaseMarshmallowConfig, ABC):
         ),
     )
 
+    def dependencies_installed(self):
+        """Some search algorithms require additional packages to be installed, check that they are available."""
+        for package_name in get_scheduler_dependencies(self.type):
+            try:
+                import_module(package_name)
+            except ImportError:
+                raise ImportError(
+                    f"Scheduler {self.type} requires package {package_name}, however package is "
+                    "not installed. Please refer to Ray Tune documentation for packages required for this "
+                    "scheduler."
```

Comment:
Does it make sense to have a ludwig requirements handy that would install all of the schedulers?

`pip install ludwig[hyperopt]`

And/or is there a link to Ray Tune documentation we can provide in the raise message here?

Reply:
Could also be useful to just say something like, `install this package by running pip install -U {package_name}`

Reply:
But would be great if they could install the full list in one go. That might be out of the scope for this PR

Reply:
I think we could register the PyPI package name as well in a tuple of `(module_name, package_name)` and raise the pip install command like that. We would have to do it like that because something like `pip install skopt` won't work. I could update like so and then catch the import error, collect the uninstalled packages, and raise the ConfigValidationError with the full command to install all missing dependencies.

Reply:
I think it would be overkill to have separate requirements for all of these search algs as well as any scheduler deps (only PBT2 for now, but that may change in the future) since most only need one additional package. Is there significant overhead to installing them all?